### PR TITLE
feat(ble): add behavior to disconnect from BLE profile

### PR DESCRIPTION
**Pull request by chrisandreae**
Adds new functionality and a behavior to disconnect an active BLE connection. The motivation for this is that for some devices like phones, the presence of an active BLE connection results in the onscreen keyboard being selected.

This behavior allows the active connection for a given profile to be explicitly disconnected, by calling [`bt_conn_disconnect()`](https://docs.zephyrproject.org/apidoc/latest/group__bt__conn.html#ga14e7c852b0271781594e742ae509c5d3). The connection will be re-established (via `update_advertising()`), when that profile is re-selected.

I believe this behavior is complementary to the proposed automatic disconnection of non-selected devices. Automatic disconnection without a timeout could be frustrating to use when actively switching between devices, as there's a slight but noticeable delay on re-establishing the connnection. On the other hand if a timeout is used, there might be times when the user wishes to disconnect before the timeout has elapsed. This behaviour would permit that.

**Questions:**

In the case that `OUT_USB` is selected, there is still a BLE profile which is considered active and selected, and is being advertised for. If the user tries to disconnect from that profile, it will be immediately re-connected from the advertisement. This would be awkward in the case that the user was wanting to switch between their computer via USB and their phone via BLE. I'm wondering if it would be most consistent to stop advertising for the selected BLE profile in the case that `OUT_USB` is selected and the USB connection is present and active. This would make selecting the USB output more consistent with selecting a different BLE output.

### CODE REVIEW

**Comment by PabloLION** [comment]
I want to specify another motivation if I may:
I always put my mac studio into lock-screen mode since it's basically sleeping mode. with this BLE function showing which device is online, the mac wakes up every once in a while. This is especially annoying when I'm taking a nap in front of the screen. It seems this PR will fix the issue.

File: app/src/ble.c
Comment (petejohanson) [inline app/src/ble.c:286]: Perhaps `ENODEV`?

File: app/src/ble.c
Comment (petejohanson) [inline app/src/ble.c:288]: Ditto

**Review by petejohanson — CHANGES_REQUESTED** [review]
One comment on the code. This also needs to update the BT behavior docs page to document the new command. See https://zmk.dev/docs/behaviors/bluetooth#bluetooth-command-defines

File: docs/docs/behaviors/bluetooth.md
Comment (caksoylar) [inline docs/docs/behaviors/bluetooth.md:42]: The length of this define kind of sticks out from the other ones. What do you think about abbreviating it, e.g. `BT_DIS` or `BT_DSC`?

File: docs/docs/behaviors/bluetooth.md
Comment (caksoylar) [inline docs/docs/behaviors/bluetooth.md:19]: I might not have understood #1998 correctly, but does it imply that this shouldn't be used with the currently active profile when USB endpoint is active? If so, can we add a warning note on that here?

File: docs/docs/behaviors/bluetooth.md
Comment (chrisandreae) [reply docs/docs/behaviors/bluetooth.md:42]: I'm torn, because those abbreviations aren't especially transparent; you'd have to check the docs to interpret them. NXT, PRV, and SEL all don't have that problem, they're quite self-explanatory.

File: docs/docs/behaviors/bluetooth.md
Comment (chrisandreae) [reply docs/docs/behaviors/bluetooth.md:19]: Well, if the disconnected profile is the currently active one, it'll successfully disconnect but then immediately reconnect, because the currently active BT profile always advertises—even if the USB output is selected. #1998 is about a follow-up to make the pause in advertising when switching away from a BT profile consistent, whether that's switching to another BT profile or switching to USB.

I mentioned it to Pete, and he said he'd prefer to leave that to a follow-up because he has a BT refactor in the works that they'd interfere with, and to add an issue to keep track of it.

I don't have a strong opinion about whether the automatic re-connection should be documented explicitly here. If the advertising behavior is changed later, the documentation can always be updated to match.

File: docs/docs/behaviors/bluetooth.md
Comment (petejohanson) [reply docs/docs/behaviors/bluetooth.md:42]: I honestly think `BT_DISC` is descriptive enough, at least as descriptive as `BT_SEL` is, and as long as it's clearly documented, should be fine.

Comment (chrisandreae) [reply docs/docs/behaviors/bluetooth.md:42]: Okay, `BT_DISC` works for me, I'll make the change

File: docs/docs/behaviors/bluetooth.md
Comment (petejohanson) [reply docs/docs/behaviors/bluetooth.md:19]: @chrisandreae I would document that expected behavior here for now, and we can update later as needed. Omitting it may just lead to some confusion on how this works *today* for the active profile.

**Review by petejohanson — COMMENTED** [review]
Code looks good to me. Happy to merge once the docs addition is in.

File: docs/docs/behaviors/bluetooth.md
Comment (chrisandreae) [reply docs/docs/behaviors/bluetooth.md]: Sure, I've done that. Does this look OK?

**Review by caksoylar — APPROVED** [review]

**Review by petejohanson — APPROVED** [review]
Thanks!

**Pull request merged**
